### PR TITLE
MGDCTRS-591 Make imagePullSecrets configurable

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/MGDCTRS-591

### CODE REVIEW

File: etc/kubernetes/operator-debezium/lgarciaac/app-config/application.properties [hunk 1]
```diff
@@ -9,4 +9,5 @@ quarkus.log.category."io.fabric8.kubernetes.client.internal.VersionUsageUtils".l
 # operator
 #
 cos.connectors.namespace = ${kubernetes.namespace}
-cos.operators.namespace   = ${kubernetes.namespace}
\ No newline at end of file
+cos.operators.namespace   = ${kubernetes.namespace}
+cos.operator.debezium.image-pull-secrets-name = "addon-pullsecret"
```
Comment (lburgazzoli) [inline etc/kubernetes/operator-debezium/lgarciaac/app-config/application.properties:13]: looks like `addon-pullsecret` is already the default value, not sure if we need tho set the value in the property files

Comment (ani-sha) [reply etc/kubernetes/operator-debezium/lgarciaac/app-config/application.properties:13]: Right, I will remove it from the property files.

**Review by lburgazzoli — APPROVED** [review]

**Review by rinaldodev — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -17,4 +17,5 @@ kamel
 .kcp/
 .kube/
 .kind/
-.mvn/mvnd.properties
\ No newline at end of file
+.mvn/mvnd.properties
+.ocm.*
```
Comment (rk3rn3r) [inline .gitignore:21]: What is this about?

Comment (ani-sha) [reply .gitignore:21]: If you install the `ocm` command-line tool, then run the `ocm login` with the provided token; a `.ocm.json` file is generated, which I have excluded here.

Comment (rk3rn3r) [reply .gitignore:21]: Thx for the clarification. Let's make this explicit then and set `.ocm.json` here.

File: etc/kubernetes/operator-debezium/lgarciaac/app-config/application.properties [hunk 3]
```diff
@@ -9,4 +9,4 @@ quarkus.log.category."io.fabric8.kubernetes.client.internal.VersionUsageUtils".l
 # operator
 #
 cos.connectors.namespace = ${kubernetes.namespace}
-cos.operators.namespace   = ${kubernetes.namespace}
\ No newline at end of file
+cos.operators.namespace   = ${kubernetes.namespace}
```
Comment (rk3rn3r) [inline etc/kubernetes/operator-debezium/lgarciaac/app-config/application.properties:12]: I am wondering about this change. What exactly changed here?
Can you maybe add a commented line with the new config property and a placeholder value?

File: cos-fleetshard-operator-debezium/src/test/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandControllerTest.java [hunk 4]
```diff
@@ -59,6 +60,11 @@
     private static final String SCHEMA_REGISTRY_ID = "9bsv0s0k8lng031se9q0";
 
     private static final DebeziumOperandConfiguration CONFIGURATION = new DebeziumOperandConfiguration() {
+        @Override
+        public LocalObjectReference imagePullSecretsName() {
+            return null;
```
Comment (rk3rn3r) [inline cos-fleetshard-operator-debezium/src/test/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandControllerTest.java:65]: Could you add another tests that comes with an actual value and 2 integration tests that test the default value and with an set value in the application.properties?

File: cos-fleetshard-operator-debezium/src/test/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandControllerTest.java [hunk 5]
```diff
@@ -208,7 +214,7 @@ void reify(ObjectNode connectorConfig, Consumer<KafkaConnect> kafkaConnectChecks
             .first()
             .isInstanceOfSatisfying(KafkaConnect.class, kc -> {
                 assertThat(kc.getSpec().getTemplate().getPod().getImagePullSecrets())
-                    .contains(DebeziumConstants.IMAGE_PULL_SECRET);
+                    .contains(CONFIGURATION.imagePullSecretsName());
```
Comment (rk3rn3r) [inline cos-fleetshard-operator-debezium/src/test/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandControllerTest.java:217]: This should not be null, but the default value (`addon-pullsecret`), shouldn't it?

**Comment by rk3rn3r** [comment]
@ani-sha did you push your latest changes? I thought we fixed those test issues?

File: pom.xml [hunk 6]
```diff
@@ -233,6 +233,11 @@
                 <artifactId>cos-fleetshard-api</artifactId>
                 <version>${cos-fleetshard-api.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.bf2</groupId>
+                <artifactId>cos-fleetshard-api-deployment</artifactId>
+                <version>${cos-fleetshard-api.version}</version>
+            </dependency>
```
Comment (rk3rn3r) [inline pom.xml:240]: Please remove this change again. #258 has a better fix, adding the dependency to the Quarkus plugin in the <plugin> section.

**Comment by ani-sha** [comment]
@rk3rn3r Yes I did, it was fixed for the unit tests & runs fine, but it failed for the ITs in the CI.

File: pom.xml (hunk 6 above)
Comment (ani-sha) [reply pom.xml:240]: Ok, sure!

**Comment by lburgazzoli** [comment]
@ani-sha needs rebase

**Comment by rk3rn3r** [comment]
Applied. Thx @ani-sha

**Pull request merged**